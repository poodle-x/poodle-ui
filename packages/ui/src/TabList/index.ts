import { TabList, TabListProps } from "./TabList";
import { StandardThemeConfig } from "../theme";

export * from "./TabList";

export default TabList;

export type TabListStyleKeys = "root";

export interface TabListThemeConfig
	extends StandardThemeConfig<TabListStyleKeys, TabListProps> {}
