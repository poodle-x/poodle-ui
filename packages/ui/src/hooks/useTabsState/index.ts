export * from "./useTabsState";
export { default } from "./useTabsState";
