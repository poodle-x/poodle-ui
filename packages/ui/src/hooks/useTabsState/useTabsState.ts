import React, { SetStateAction } from "react";

export type TabValue = string | number;

export enum TabRegisterType {
	TAB = "TAB",
	PANEL = "PANEL",
}

export enum TabOrientation {
	VERTICAL = "vertical",
	HORIZONTAL = "horizontal",
}

export interface TabsContextValue {
	orientation: TabOrientation;
	listTab: [TabValue, string][];
	listTabPanel: [TabValue, string][];
	value?: TabValue;
	onChange?: (
		value: TabValue,
		event: React.SyntheticEvent | KeyboardEvent
	) => void;
	register: (
		type: TabRegisterType,
		value: TabValue,
		id: string,
		index: number
	) => void;
	unregister: (type: TabRegisterType, value: TabValue, id: string) => void;
}

export interface UseTabsStateOptions {
	propValue?: TabValue;
}

export const TabsContext = React.createContext<TabsContextValue>(
	{} as TabsContextValue
);

export default function useTabsState(props: UseTabsStateOptions) {
	const { propValue } = props;
	const [unControlledValue, setUnControlledValue] = React.useState<
		TabValue | undefined
	>();
	const [listTab, setListTab] = React.useState<[TabValue, string][]>([]);
	const [listPanel, setListPanel] = React.useState<[TabValue, string][]>([]);

	const handleRegister: TabsContextValue["register"] = React.useCallback(
		(type, value, id, index) => {
			if (type === TabRegisterType.TAB && index !== undefined) {
				setListTab((lt) => {
					return [...lt.slice(0, index), [value, id], ...lt.slice(index)];
				});
				return;
			}

			if (type === TabRegisterType.PANEL && index !== undefined) {
				setListPanel((lt) => {
					return [...lt.slice(0, index), [value, id], ...lt.slice(index)];
				});
				return;
			}

			const action: SetStateAction<[TabValue, string][]> = (cs) => {
				const current = cs.findIndex((c) => c[0] === value);
				if (current === -1) {
					return [...cs, [value, id]];
				} else {
					return cs.map((c, i) => {
						if (i !== current) {
							return c;
						} else {
							return [value, id];
						}
					});
				}
			};
			if (type === TabRegisterType.TAB) {
				setListTab(action);
			} else if (type === TabRegisterType.PANEL) {
				setListPanel(action);
			}
		},
		[]
	);

	const handleUnregister: TabsContextValue["unregister"] = React.useCallback(
		(type, value) => {
			const action: SetStateAction<[TabValue, string][]> = (cs) => {
				const current = cs.findIndex((c) => c[0] === value);
				if (current !== -1) {
					return cs.reduce<[TabValue, string][]>((prev, cur, index) => {
						if (index !== current) {
							prev.push(cur);
						}
						return prev;
					}, []);
				}
				return cs;
			};
			if (type === TabRegisterType.TAB) {
				setListTab(action);
			} else if (type === TabRegisterType.PANEL) {
				setListPanel(action);
			}
		},
		[]
	);

	React.useEffect(() => {
		if (
			propValue === undefined &&
			unControlledValue === undefined &&
			listTab.length > 0
		) {
			setUnControlledValue(listTab[0][0]);
		}
	}, [propValue, listTab, unControlledValue]);

	return {
		setUnControlledValue,
		value: propValue === undefined ? unControlledValue : propValue,
		isUncontrolled: propValue === undefined,
		listTab,
		listTabPanel: listPanel,
		handleUnregister,
		handleRegister,
	};
}
