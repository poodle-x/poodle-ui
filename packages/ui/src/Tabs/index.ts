import { Tabs, TabsProps } from "./Tabs";
import { StandardThemeConfig } from "../theme";
import { TabValue, TabsContext, TabsContextValue } from "../hooks/useTabsState";

export * from "./Tabs";

export default Tabs;

export type TabsStyleKeys = "root" | "list";

export interface TabsThemeConfig
	extends StandardThemeConfig<TabsStyleKeys, TabsProps> {}

export { TabValue, TabsContext, TabsContextValue };
