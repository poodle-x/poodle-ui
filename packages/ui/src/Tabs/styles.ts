import { createCSSSystemStandard } from "../theme";
import { CSSSystem } from "../styled";
import { TabsProps } from "./Tabs";
import { TabsStyleKeys } from "./index";
import { TabOrientation } from "../hooks/useTabsState";

export function Root(props: TabsProps): CSSSystem {
	const { theme, orientation = TabOrientation.HORIZONTAL } = props;

	return createCSSSystemStandard<TabsStyleKeys>({
		key: "root",
		config: theme?.Tabs,
		props,
		base: [
			{
				display: "flex",
				flexDirection:
					orientation === TabOrientation.VERTICAL ? "row" : "column",
			},
		],
	});
}
