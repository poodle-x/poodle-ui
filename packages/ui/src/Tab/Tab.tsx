import React from "react";
import Box, { BoxProps } from "../Box";
import useTab from "../hooks/useTab";
import { TabsContext } from "../hooks/useTabsState";
import { getCSSSystemBoxProps } from "../styled/CSSSystem";
import { StandardComponentProps, ThemeConfig } from "../theme";
import uSetRef from "../utils/setRef";
import useDefaultProps from "../hooks/useDefaultProps/useDefaultProps";
import * as styles from "./styles";

export interface LocalTabProps {
	children?: React.ReactNode;
	value: string | number;
}

export interface TabProps
	extends Omit<BoxProps, "value">,
		StandardComponentProps,
		LocalTabProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.Tab?.defaultProps;
}

export const Tab: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<TabProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, TabProps>((_props, ref) => {
	const { props, isLocalTheme } = useDefaultProps<TabProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const tabsContext = React.useContext(TabsContext);

	const { children, className, value, onClick, id, ...otherProps } = props;

	const { props: htmlProps, setRef } = useTab({
		id,
		value,
	});

	function handleClick(e: React.SyntheticEvent) {
		tabsContext.onChange?.(value, e);
		e.persist && e.persist();
		onClick && onClick(e as any);
	}

	const composeRef = React.useCallback(
		(node: HTMLElement) => {
			setRef(node);
			uSetRef(ref, node);
		},
		[setRef, ref]
	);

	return (
		<Box
			data-orientation={tabsContext.orientation}
			as="button"
			{...htmlProps}
			{...otherProps}
			{...getCSSSystemBoxProps({
				isRoot: true,
				isLocalTheme,
				componentProps: props,
				fnCSSSystem: styles.Root,
				baseClassName: ["poodle-tab"],
			})}
			onClick={handleClick}
			ref={composeRef}
		>
			{children}
		</Box>
	);
});

Tab.displayName = "PoodleTab";
