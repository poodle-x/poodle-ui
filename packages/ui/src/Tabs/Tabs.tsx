import React from "react";
import Box, { BoxProps } from "../Box";
import useDefaultProps from "../hooks/useDefaultProps/useDefaultProps";
import useTabsState, {
	TabOrientation,
	TabsContext,
	TabsContextValue,
	TabValue,
} from "../hooks/useTabsState";
import { getCSSSystemBoxProps } from "../styled/system";
import { StandardComponentProps, ThemeConfig } from "../theme";
import * as styles from "./styles";

export { TabOrientation, TabsContext, TabsContextValue, TabValue };

export interface LocalTabsProps {
	/**
	 * Send value to control active tab. By default <Tabs /> component
	 * use internal state to control active tab.
	 */
	value?: TabValue;
	/**
	 * Change tabs horizontal orientation or vertical orientation.
	 * Default is horizontal orientation.
	 */
	orientation?: TabOrientation;
	onChangeValue?: (
		value: TabValue,
		event: React.SyntheticEvent | KeyboardEvent
	) => void;
	children?: React.ReactNode;
}

export interface TabsProps
	extends Omit<BoxProps, "value">,
		StandardComponentProps,
		LocalTabsProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.Tabs?.defaultProps;
}

export const Tabs: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<TabsProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, TabsProps>((_props, ref) => {
	const { props, isLocalTheme } = useDefaultProps<TabsProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const {
		children,
		className,
		value,
		onChangeValue,
		orientation,
		...otherProps
	} = props;

	const {
		setUnControlledValue,
		isUncontrolled,
		value: calcValue,
		handleUnregister,
		handleRegister,
		listTabPanel,
		listTab,
	} = useTabsState({ propValue: value });

	const handleChangeValue: TabsContextValue["onChange"] = React.useCallback(
		(v: TabValue, e: React.SyntheticEvent<Element, Event> | KeyboardEvent) => {
			onChangeValue && onChangeValue(v, e);
			if (isUncontrolled) {
				setUnControlledValue(v);
			}
		},
		[onChangeValue, isUncontrolled, setUnControlledValue]
	);

	return (
		<TabsContext.Provider
			value={{
				listTab,
				listTabPanel,
				value: calcValue,
				register: handleRegister,
				unregister: handleUnregister,
				onChange: handleChangeValue,
				orientation: orientation || TabOrientation.HORIZONTAL,
			}}
		>
			<Box
				{...otherProps}
				{...getCSSSystemBoxProps({
					isRoot: true,
					isLocalTheme,
					componentProps: props,
					fnCSSSystem: styles.Root,
					baseClassName: ["poodle-tabs"],
				})}
				ref={ref}
			>
				{children}
			</Box>
		</TabsContext.Provider>
	);
});

Tabs.displayName = "PoodleTabs";
