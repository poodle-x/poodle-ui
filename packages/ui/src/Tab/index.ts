import { Tab, TabProps } from "./Tab";
import { StandardThemeConfig } from "../theme";

export * from "./Tab";

export default Tab;

export type TabStyleKeys = "root";

export interface TabThemeConfig
	extends StandardThemeConfig<TabStyleKeys, TabProps> {}
