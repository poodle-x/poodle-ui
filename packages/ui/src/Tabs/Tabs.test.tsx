import { axe } from "jest-axe";
import React from "react";
import { TabOrientation } from "../hooks/useTabsState";
import { Tab } from "../Tab";
import { TabList } from "../TabList";
import { TabPanel } from "../TabPanel";
import { ThemeConfig } from "../theme";
import { fireEvent, render } from "../utils/test";
import Tabs from "./";

describe("<Tabs />", () => {
	it("should render Tabs correctly", () => {
		const { container } = render(<Tabs>Tabs</Tabs>);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("ref should work", () => {
		const ref = React.createRef<HTMLDivElement>();

		render(<Tabs ref={ref}>Tabs</Tabs>);
		expect(ref.current).toMatchSnapshot();
	});

	it("should render Box props correctly", () => {
		const { container } = render(
			<Tabs id="id" mt="20px">
				Tabs
			</Tabs>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	function checkTabActive(tab: HTMLElement) {
		expect(tab.getAttribute("aria-selected")).toEqual("true");
		expect(tab.getAttribute("tabindex")).toEqual("0");
	}

	function checkTabInActive(tab: HTMLElement) {
		expect(tab.getAttribute("aria-selected")).toEqual("false");
		expect(tab.getAttribute("tabindex")).toEqual("-1");
	}

	function checkTabPanelActive(tab: HTMLElement) {
		expect(tab.getAttribute("hidden")).toEqual(null);
	}

	function checkTabPanelInActive(tab: HTMLElement) {
		expect(tab.getAttribute("hidden")).toEqual("");
	}

	function checkTabs(
		tabs: HTMLElement[],
		tabPanels: HTMLElement[],
		active: number,
		inactive: number[]
	) {
		checkTabActive(tabs[active]);
		checkTabPanelActive(tabPanels[active]);

		inactive.forEach((i) => {
			checkTabInActive(tabs[i]);
			checkTabPanelInActive(tabPanels[i]);
		});
	}

	const renderKey = (ori: TabOrientation) => {
		return (
			<Tabs orientation={ori}>
				<TabList aria-label="Settings">
					<Tab value={0}>First</Tab>
					<Tab value={1}>Second</Tab>
					<Tab value={2}>Third</Tab>
				</TabList>
				<TabPanel value={0}>First</TabPanel>
				<TabPanel value={1}>Second</TabPanel>
				<TabPanel value={2}>Third</TabPanel>
			</Tabs>
		);
	};

	it("should pass axe", async () => {
		const { container } = render(renderKey(TabOrientation.HORIZONTAL));
		const results = await axe(container);
		expect(results).toHaveNoViolations();
	});

	it("should change active tab uncontrolled", () => {
		const { getAllByRole } = render(renderKey(TabOrientation.HORIZONTAL));

		const tabs = getAllByRole("tab");

		const tabPanels = getAllByRole("tabpanel", {
			hidden: true,
		});

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.click(tabs[1]);

		checkTabs(tabs, tabPanels, 1, [0, 2]);

		fireEvent.click(tabs[2]);

		checkTabs(tabs, tabPanels, 2, [0, 1]);
	});

	it("should change active tab horizontal with keyboard arrow", () => {
		const { getAllByRole, getByRole } = render(
			renderKey(TabOrientation.HORIZONTAL)
		);

		const tabList = getByRole("tablist");

		const tabs = getAllByRole("tab");

		const tabPanels = getAllByRole("tabpanel", {
			hidden: true,
		});

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowRight", code: "ArrowRight" });

		checkTabs(tabs, tabPanels, 1, [0, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowRight", code: "ArrowRight" });

		checkTabs(tabs, tabPanels, 2, [0, 1]);

		fireEvent.keyDown(tabList, { key: "ArrowRight", code: "ArrowRight" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowLeft", code: "ArrowLeft" });

		checkTabs(tabs, tabPanels, 2, [0, 1]);

		fireEvent.keyDown(tabList, { key: "ArrowLeft", code: "ArrowLeft" });

		checkTabs(tabs, tabPanels, 1, [0, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowLeft", code: "ArrowLeft" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.keyDown(tabList, { key: "End", code: "End" });

		checkTabs(tabs, tabPanels, 2, [0, 1]);

		fireEvent.keyDown(tabList, { key: "Home", code: "Home" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		// Shouldn't do anything
		fireEvent.keyDown(tabList, { key: "ArrowUp", code: "ArrowUp" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowDown", code: "ArrowDown" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);
	});

	it("should change active tab vertical with keyboard arrow", () => {
		const { getAllByRole, getByRole } = render(
			renderKey(TabOrientation.VERTICAL)
		);

		const tabList = getByRole("tablist");

		const tabs = getAllByRole("tab");

		const tabPanels = getAllByRole("tabpanel", {
			hidden: true,
		});

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowDown", code: "ArrowDown" });

		checkTabs(tabs, tabPanels, 1, [0, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowDown", code: "ArrowDown" });

		checkTabs(tabs, tabPanels, 2, [0, 1]);

		fireEvent.keyDown(tabList, { key: "ArrowDown", code: "ArrowDown" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowUp", code: "ArrowUp" });

		checkTabs(tabs, tabPanels, 2, [0, 1]);

		fireEvent.keyDown(tabList, { key: "ArrowUp", code: "ArrowUp" });

		checkTabs(tabs, tabPanels, 1, [0, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowUp", code: "ArrowUp" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.keyDown(tabList, { key: "End", code: "End" });

		checkTabs(tabs, tabPanels, 2, [0, 1]);

		fireEvent.keyDown(tabList, { key: "Home", code: "Home" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		// Shouldn't do anything
		fireEvent.keyDown(tabList, { key: "ArrowLeft", code: "ArrowLeft" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowRight", code: "ArrowRight" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);
	});

	it("should change active tab when have disabled tabs", () => {
		const { getAllByRole, getByRole, rerender } = render(
			<Tabs>
				<TabList aria-label="Settings">
					<Tab value={0}>First</Tab>
					<Tab value={1} disabled={true}>
						Second
					</Tab>
					<Tab value={2}>Third</Tab>
				</TabList>
				<TabPanel value={0}>First</TabPanel>
				<TabPanel value={1}>Second</TabPanel>
				<TabPanel value={2}>Third</TabPanel>
			</Tabs>
		);

		let tabList = getByRole("tablist");

		let tabs = getAllByRole("tab");

		let tabPanels = getAllByRole("tabpanel", {
			hidden: true,
		});

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		fireEvent.keyDown(tabList, { key: "ArrowRight", code: "ArrowRight" });

		checkTabs(tabs, tabPanels, 2, [0, 1]);

		fireEvent.keyDown(tabList, { key: "ArrowLeft", code: "ArrowLeft" });

		checkTabs(tabs, tabPanels, 0, [1, 2]);

		rerender(
			<Tabs>
				<TabList aria-label="Settings">
					<Tab value={0} disabled={true}>
						First
					</Tab>
					<Tab value={1}>Second</Tab>
					<Tab value={2}>Third</Tab>
					<Tab value={3} disabled={true}>
						Fourth
					</Tab>
				</TabList>
				<TabPanel value={0}>First</TabPanel>
				<TabPanel value={1}>Second</TabPanel>
				<TabPanel value={2}>Third</TabPanel>
				<TabPanel value={3}>Fourth</TabPanel>
			</Tabs>
		);

		tabList = getByRole("tablist");

		tabs = getAllByRole("tab");

		tabPanels = getAllByRole("tabpanel", {
			hidden: true,
		});

		fireEvent.keyDown(tabList, { key: "End", code: "End" });

		checkTabs(tabs, tabPanels, 2, [0, 1, 3]);

		fireEvent.keyDown(tabList, { key: "Home", code: "Home" });

		checkTabs(tabs, tabPanels, 1, [0, 2, 3]);
	});

	describe("theming", () => {
		const themeTest: ThemeConfig = {
			mode: "mode",
			Tabs: {
				styles: {
					root: {
						mt: 10,
						letterSpacing: 250,
					},
				},
				variants: {
					custom: {
						root: {
							top: 0,
							textIndent: 1,
						},
					},
				},
			},
		};

		it("should extend styles in global", () => {
			const { container } = render(<Tabs variant="custom">Tabs</Tabs>, {
				theme: themeTest,
			});
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should extend styles in local", () => {
			const { container } = render(
				<Tabs variant="custom" themeExtend={themeTest}>
					Tabs
				</Tabs>
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should render with default props correctly", () => {
			const { container } = render(<Tabs>Tabs</Tabs>, {
				theme: {
					Tabs: {
						defaultProps: {
							id: "id",
							className: "hello",
						},
					},
				},
			});
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should override styles", () => {
			const { container } = render(<Tabs>Tabs</Tabs>, {
				theme: {
					Tabs: {
						overrides: {
							root: {
								mt: 50,
							},
						},
					},
				},
			});
			expect(container.firstChild).toMatchSnapshot();
		});
	});
});
