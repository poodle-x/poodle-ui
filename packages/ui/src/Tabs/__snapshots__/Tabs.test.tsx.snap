// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> ref should work 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="poodle-box poodle-tabs c0"
>
  Tabs
</div>
`;

exports[`<Tabs /> should render Box props correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 20px;
}

<div
  class="poodle-box poodle-tabs c0"
  id="id"
>
  Tabs
</div>
`;

exports[`<Tabs /> should render Tabs correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="poodle-box poodle-tabs c0"
>
  Tabs
</div>
`;

exports[`<Tabs /> theming should extend styles in global 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

<div
  class="poodle-box poodle-tabs c0"
>
  Tabs
</div>
`;

exports[`<Tabs /> theming should extend styles in local 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

<div
  class="poodle-box poodle-tabs c0"
>
  Tabs
</div>
`;

exports[`<Tabs /> theming should override styles 1`] = `
.c0 {
  margin-top: 50px;
}

<div
  class="poodle-box poodle-tabs c0"
>
  Tabs
</div>
`;

exports[`<Tabs /> theming should render with default props correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="poodle-box poodle-tabs hello c0"
  id="id"
>
  Tabs
</div>
`;
